src/nvim/os/env.c

```diff
@@ -300,6 +303,36 @@ void init_homedir(void)
   if (var == NULL) {
     var = os_getenv("USERPROFILE");
   }
+
+  // Weird but true: $HOME may contain an indirect reference to another
+  // variable, esp. "%USERPROFILE%".  Happens when $USERPROFILE isn't set
+  // when $HOME is being set.
+  if (var != NULL && *var == '%') {
+    char *p;
+    const char *exp;
+
+    p = strchr(var + 1, '%');
+    if (p != NULL) {
+      vim_snprintf(os_buf, (size_t)(p - var), "%s", var + 1);
+      exp = os_getenv(os_buf);
+      if (exp != NULL && *exp != NUL
+          && STRLEN(exp) + STRLEN(p) < MAXPATHL) {
+        vim_snprintf(os_buf, MAXPATHL, "%s%s", exp, p + 1);
+        var = os_buf;
+      }
+    }
+  }
+
+  if (var != NULL && *var == NUL) {
+    // empty is same as not set
+    var = NULL;
+  }
+
+  // Default home dir is C:/
+  // Best assumption we can make in such a situation.
+  if (var == NULL) {
+    var = "C:/";
+  }
```

Comment:
this could be `if (var == NULL || *var == NUL)`, then the block at line 326 could be removed.